python/cudf/cudf/core/frame.py

```diff
@@ -3301,6 +3301,65 @@ def _encode(self):
         keys = self.__class__._from_table(keys)
         return keys, indices
 
+    def _reindex_frame(self, columns, dtypes=None, deep=False, index=None):
```

Comment:
`Frame._reindex_frame` is redundant, let's just call it `_reindex`. Apologies, my earlier suggestion of `_reindex_frame` assumed that it was going to be a free function.